### PR TITLE
DRA: test checkpoint structure upgrade

### PR DESCRIPTION
**Pull request by bart0sh**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Added new test case that emulates upgrade scenario: adding new field to the checkpointed structure and loading checkpoint that was made before the structure was changed.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/dra/state/checkpointer_test.go [hunk 1]
```diff
@@ -148,6 +148,32 @@ func TestCheckpointGetOrCreate(t *testing.T) {
 			checkpointContent: `{`,
 			expectedError:     "unexpected end of JSON input",
 		},
+		{
+			// Emulate upgrade scenario: addition of the `Device.RequestNames`` field
+			// Checkpoint content doesn't have RequestNames field, but it shouldn't prevent the checkpoint from being loaded
```
Comment (pohly) [inline pkg/kubelet/cm/dra/state/checkpointer_test.go:153]:
```suggestion
			// Emulate upgrade scenario by pretending that `Device.RequestNames` is a new field.
			// The checkpoint content doesn't have it and that shouldn't prevent the checkpoint from being loaded.
```

**Review by pohly — CHANGES_REQUESTED** [review]

**Review by pohly — APPROVED** [review]
/lgtm
/assign @klueska 

For approval.

**Comment by bart0sh** [comment]
/assign @mrunalp @SergeyKanzhelev 
for approval

**Comment by bart0sh** [comment]
@klueska I've added one more test case with updated checkpoint version. PTAL.

**Comment by klueska** [comment]
Does the new test case only work currently because we don't actually check the version at all and attempt to do any sort of migrations? So we basically put it there so that it will fail if / when we add such support to make sure we are doing things correctly?

**Comment by bart0sh** [comment]
@klueska Yes, that's how it's implemented.

**Pull request merged**